packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/MrwSerializationTypeDefinitions/MrwSerializationTypeDefinitionTests.cs

```diff
@@ -585,5 +586,50 @@ public void TestBuildExplicitFromClientResult()
             var methodBody = method?.BodyStatements;
             Assert.IsNotNull(methodBody);
         }
+
+        [Test]
+        public void TestIntSerializationStatement(
+            [Values(
+            InputPrimitiveTypeKind.Integer,
+            InputPrimitiveTypeKind.SafeInt,
+            InputPrimitiveTypeKind.Int8,
+            // InputPrimitiveTypeKind.Int16, TODO: add them back when we decide the exact corresponding type
+            InputPrimitiveTypeKind.Int32,
+            InputPrimitiveTypeKind.Int64,
+            InputPrimitiveTypeKind.UInt8
+            // InputPrimitiveTypeKind.UInt16,
+            // InputPrimitiveTypeKind.UInt32,
+            // InputPrimitiveTypeKind.UInt64
+            )] InputPrimitiveTypeKind kind,
+            [Values("string", null)] string encode)
+        {
+            var name = kind.ToString().ToLower();
+            var properties = new List<InputModelProperty>
+            {
+                new InputModelProperty("requiredInt", "requiredInt", "", new InputPrimitiveType(kind, name, $"TypeSpec.{name}", encode), true, false, false),
+             };
+
+            var inputModel = new InputModelType("TestModel", "TestModel", "public", null, "Test model.", InputModelTypeUsage.Input, properties, null, Array.Empty<InputModelType>(), null, null, new Dictionary<string, InputModelType>(), null, false);
+
+            var (_, serialization) = CreateModelAndSerialization(inputModel);
+
+            Assert.IsTrue(TestHelpers.HasExpressionStatement(serialization.BuildJsonModelWriteCoreMethod().BodyStatements, encode is null ? "writer.WriteNumberValue(RequiredInt);\n" : "writer.WriteStringValue(RequiredInt.ToString());\n"));
+        }
```

Comment:
Could we just use an asset file to assert the result?
In this way we do not have to use the `HasExpressionStatement` method which is so complicated to use.
Here is an example:
https://github.com/microsoft/typespec/blob/71c221d4d91da2e0dc40d114e7455ea524dafa3a/packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/MrwSerializationTypeDefinitions/JsonModelCoreTests.cs#L16

Reply:
There is cadl-ranch test which actually does the same work. (I'll add it later).
That's why I don't use assert file. My thought is that unit test can run fast and it's easier to debug.

Reply:
Another lesson I learnt in `autorest.csharp` is that comparing generated files are harder to maintain in long term. We have left some bugs before, just because no one can really check every line of change.